From 71ea6aba1d7f1878171919b9ec12f42b791953e1 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 18 Oct 2018 18:35:03 +0300
Subject: [PATCH 0720/1051] ARM64: dts: marvell: Fix the PCIe x4 PHY
 configuration

Fix the PHY's list in A7K and A8K DB boards setup "B" running
PCIe0 on CPo in x4 mode (4 SERDES lanes)

Change-Id: I970347116e6c60c2a728ab4f9714c932a8e54c1a
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60390
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-B.dts | 6 +++++-
 arch/arm64/boot/dts/marvell/armada-8040-db-B.dts | 6 +++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
index 805d9e16809f..709212035a69 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
@@ -42,7 +42,11 @@
 &cp0_pcie0 {
 	status = "okay";
 	num-lanes = <4>;
-	phys = <&cp0_comphy0 0>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy0 0
+		&cp0_comphy1 0
+		&cp0_comphy2 0
+		&cp0_comphy3 0>;
 };
 
 &cp0_pcie2 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
index 4638e2bb955c..25f40a545f2b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
@@ -44,7 +44,11 @@
 &cp0_pcie0 {
 	num-lanes = <4>;
 	status = "okay";
-	phys = <&cp0_comphy0 0>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy0 0
+		&cp0_comphy1 0
+		&cp0_comphy2 0
+		&cp0_comphy3 0>;
 };
 
 /* CON5 on CP0 expansion */
-- 
2.17.1

